src/adapter/src/catalog.rs

```diff
@@ -1453,6 +1453,7 @@ pub struct Table {
     pub defaults: Vec<Expr<Aug>>,
     pub conn_id: Option<ConnectionId>,
     pub depends_on: Vec<GlobalId>,
+    pub custom_logical_compaction_window: Option<Duration>,
```

Comment:
We did consider this (with @jkosh44); however, we realized that if we open this to users, and a user manually sets 1s (the same value as the default), then if we ever want to change the default, we will not be able to tell whether tables with 1s compaction window should be changed (because they were just inheriting the default) or left with 1s (because a user set them).

So I decided to leave this as optional, to communicate that things with `None` should inherit whatever the default is, including if it's updated to something other than 1s.

Reply:
Okay, that makes sense!